### PR TITLE
Fixing deprecations for Shapely 2.0, Numpy and Pyproj

### PR DESCRIPTION
**Pull request by ekaats**
I have been adding some fixes to support Shapely 2.0 and fix some future deprecations around pyproj and numpy.

There is still some weird stuff going on, particularly with providers that don't use 3857. I've mostly been testing with PDOK (Netherlands) as a provider, which does use 3857, but used the non-standard routine anyway. I fixed that, but need to do more testing with non-standard providers.

It takes some time to figure out if these are regressions or not.

### CODE REVIEW

**Comment by ekaats** [comment]
Couldn't reproduce the earlier issues. Tested with PDOK and global providers and I don't get any errors.

I do realise I changed out the Nvidia converter for Imagemagick, at least for Linux. I couldn't find the library for my distro and a short search told me that it is not supported anymore. I'll leave it to someone else to decide if that is a welcome change.

**Comment by anthraxn8b** [comment]
Good idea to introduce the "build" directory!

**Comment by anthraxn8b** [comment]
Added my PR on top of @ekaats fork: https://github.com/ekaats/Ortho4XP/pull/1

**Comment by anthraxn8b** [comment]
@ekaats, after reading though the issues I found out, that some people tried/try(?) to make a community software out of this. Maybe it would make sense to do the same work again in the community product?
https://github.com/Ortho4XP/Ortho4XP (in the DEV branch!)

**Comment by kyuknis** [comment]
Tried it today and it seems mostly good to go. I moved all the tile directories over but where are the overlays to move into custom scenery?

**Comment by d41k4n** [comment]
> where are the overlays to move into custom scenery?

Overlays are extracted into folder "yOrtho4XP_Overlays" within the Ortho4XP home folder. Btw. the official support forum is located [here](https://forums.x-plane.org/index.php?/forums/forum/322-ortho4xp/).

**Review by w8sl — APPROVED** [review]
Brilliant update - thank You!

**Comment by jonaseberle** [comment]
Now that we have a `requirement.txt` this Pull Request would just need to update it.

I am happy to test. Any region/provider to look out for?

**Comment by w8sl** [comment]
EPSG3857 The most common CRS for online maps, used by almost all free and commercial tile providers ?

File: src/O4_Imagery_Utils.py [hunk 1]
```diff
@@ -922,7 +906,15 @@ def build_texture_from_bbox_and_size(t_bbox,t_epsg,t_size,provider):
...
-        UI.vprint(3,"Resize needed:"+str(t_size[0]/big_image.size[0])+" "+str(t_size[1]/big_image.size[1]))
+        try:
+            UI.vprint(3,"Resize needed:"+str(t_size[0]/big_image.size[0])+" "+str(t_size[1]/big_image.size[1]))
+        except ZeroDivisionError as e:
+            # Something is going wrong here. Non-epsg-3857 may be impacted but apart from the print not working
+            # the following resize doesn't actually throw an error.
+            print("division by zero at build_texture_from_bbox_and_size")
+            pass
+        else:
+            print("works")
```
Comment (d41k4n) [inline src/O4_Imagery_Utils.py:917]:
```suggestion
            UI.vprint(3,"works")
```

File: src/O4_Imagery_Utils.py [hunk 2]
```diff
@@ -922,7 +906,15 @@ def build_texture_from_bbox_and_size(t_bbox,t_epsg,t_size,provider):
         UI.vprint(3,"Crop needed")
         big_image=big_image.crop((crop_x0,crop_y0,crop_x1,crop_y1))
     if big_image.size!=t_size:
-        UI.vprint(3,"Resize needed:"+str(t_size[0]/big_image.size[0])+" "+str(t_size[1]/big_image.size[1]))
+        try:
+            UI.vprint(3,"Resize needed:"+str(t_size[0]/big_image.size[0])+" "+str(t_size[1]/big_image.size[1]))
+        except ZeroDivisionError as e:
+            # Something is going wrong here. Non-epsg-3857 may be impacted but apart from the print not working
+            # the following resize doesn't actually throw an error.
+            print("division by zero at build_texture_from_bbox_and_size")
```
Comment (d41k4n) [inline src/O4_Imagery_Utils.py:914]:
```suggestion
            UI.vprint(3,"division by zero at build_texture_from_bbox_and_size")
```

**Review by d41k4n — CHANGES_REQUESTED** [review]
Probably better to use the existing vprint function for debug logging to be more consistent.

**Comment by d41k4n** [comment]
@jonaseberle 
> Now that we have a `requirement.txt` this Pull Request would just need to update it.

I'd suggest to either rebase this branch on the latest master or merge first and then update the requirements.txt with a separate PR.

**Comment by jonaseberle** [comment]
@d41k4n both is fine, thanks for taking care.

**Review by w8sl — CHANGES_REQUESTED** [review]
Coordinates for non EPSG3857 providers are wrong.
Fix:
line 26 in src/04_GEO_Utils.py:
return Transformer.from_crs(crs_from=f"EPSG:{s_epsg}", crs_to=f"EPSG:{t_epsg}", always_xy=True).transform(s_x, s_y)

File: src/O4_Geo_Utils.py [hunk 3]
```diff
@@ -15,19 +16,15 @@ def dist(A,B):
...
-
 ##############################################################################
 def webmercator_pixel_size(lat,zoomlevel):
     return 2*pi*earth_radius*cos(pi*lat/180)/(2**(zoomlevel+8))
 ##############################################################################
 
 ##############################################################################
 def transform(s_epsg, t_epsg, s_x, s_y):
-    return pyproj.transform(epsg[s_epsg], epsg[t_epsg], s_x, s_y)
+    return Transformer.from_crs(crs_from=f"EPSG:{s_epsg}", crs_to=f"EPSG:{t_epsg}").transform(s_x, s_y)
```
Comment (d41k4n) [inline src/O4_Geo_Utils.py:26]: As per @w8sl 's suggestion:
```suggestion
    return Transformer.from_crs(crs_from=f"EPSG:{s_epsg}", crs_to=f"EPSG:{t_epsg}", always_xy=True).transform(s_x, s_y)
```

**Review by d41k4n — CHANGES_REQUESTED** [review]

**Review by w8sl — CHANGES_REQUESTED** [review]
Fix for deprecation affecting road_level>2:
Line 200 in 04_Vector_Utils.py:
`    road_network_banked=geometry.MultiLineString(list(road_network_banked.geoms)+list(road_network_banked_2.geoms))`

File: src/O4_File_Names.py [hunk 4]
```diff
@@ -6,20 +6,23 @@
...
+Filter_dir = os.path.join(Ortho4XP_dir, "Filters")
+OSM_dir = os.path.join(Ortho4XP_dir, build_dir, "OSM_data")
+Mask_dir = os.path.join(Ortho4XP_dir, build_dir, "Masks")
+Imagery_dir = os.path.join(Ortho4XP_dir, build_dir, "Orthophotos")
+Elevation_dir = os.path.join(Ortho4XP_dir, build_dir, "Elevation_data")
+Geotiff_dir = os.path.join(Ortho4XP_dir, build_dir, "Geotiffs")
+Patch_dir = os.path.join(Ortho4XP_dir, "Patches")
+Utils_dir = os.path.join(Ortho4XP_dir, "Utils")
+Tile_dir = os.path.join(Ortho4XP_dir, build_dir, "Tiles")
+Tmp_dir = os.path.join(Ortho4XP_dir, build_dir, "tmp")
```
Comment (w8sl) [inline src/O4_File_Names.py:24]: "tmp" seems to be hardcoded to main folder. Exotic stuff like "imprint_masks_to_dds" but still.

Tmp_dir       =  os.path.join(Ortho4XP_dir, 'tmp')

**Pull request merged**